### PR TITLE
Scala filetype support

### PR DESCRIPTION
**Pull request by derekwyatt**
- Added hook to recognize `.scala` files
- Added `scala` specific configurations via `ftplugin` file
- Added `scala` specific syntax
- Added `scala` specific indentation

### CODE REVIEW

**Comment by chrisbra** [comment]
included as of e4a3bcf28d92d0bde9ca227ccb40

File: runtime/ftplugin/scala.vim [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+
+setlocal shiftwidth=2 softtabstop=2 expandtab
+
+setlocal include='^\s*import'
+setlocal includeexpr='substitute(v:fname,"\\.","/","g")'
+
+setlocal path+=src/main/scala,src/test/scala
+setlocal suffixesadd=.scala
+
+compiler sbt
```
Comment (itchyny) [inline runtime/ftplugin/scala.vim:35]: This line yields `E666: compiler not supported: sbt`.

Comment (derekwyatt) [reply runtime/ftplugin/scala.vim:35]: Sorry, I don't use SBT integration with Vim.  Do you think this should be removed?

Comment (brammool) [inline runtime/ftplugin/scala.vim:35]: Derek Wyatt wrote:

> > +if get(g:, 'scala_scaladoc_indent', 0)
> > -  setlocal comments=sO:_\ -,mO:_\ \ ,exO:_/,s2:/__,mb:_,ex:_/,s1:/_,mb:_,ex:_/,://
> >   +else
> > -  setlocal comments=sO:_\ -,mO:_\ \ ,exO:_/,s1:/__,mb:_,ex:_/,s1:/_,mb:_,ex:_/,://
> >   +endif
> >   +setlocal commentstring=//\ %s
> >   +
> >   +setlocal shiftwidth=2 softtabstop=2 expandtab
> >   +
> >   +setlocal include='^\s*import'
> >   +setlocal includeexpr='substitute(v:fname,"\.","/","g")'
> >   +
> >   +setlocal path+=src/main/scala,src/test/scala
> >   +setlocal suffixesadd=.scala
> >   +
> >   +compiler sbt
> 
> Sorry, I don't use SBT integration with Vim.  Do you think this should
> be removed?

Yes.  A compiler should only be set in the ftplugin if most people use
it.  E.g. for Java you can expect most people to use "javac".

## 

hundred-and-one symptoms of being an internet addict:
174. You know what a listserv is.

 /// Bram Moolenaar -- Bram@Moolenaar.net -- http://www.Moolenaar.net   \\
///        sponsor Vim, vote for features -- http://www.Vim.org/sponsor/ \\
\\  an exciting new programming language -- http://www.Zimbu.org        ///
 \\            help me help AIDS victims -- http://ICCF-Holland.org    ///

Comment (itchyny) [reply runtime/ftplugin/scala.vim:35]: @derekwyatt If you included compiler/sbt.vim in this patch, there would not be a problem. Did you test the patch without your scala plugin? Well, agree for removal anyway.

Comment (derekwyatt) [reply runtime/ftplugin/scala.vim:35]: I'm afraid I only tested in "my" environment, which isn't the smartest thing to do.  I'll retest this in something cleaner and then make the modification.  I think the removal makes sense anyway - incorporating SBT into Vim really feels like "anti-Vim" to me.  Bootstrapping and running SBT is expensive and works far better outside of Vim as a continually running process.

**Comment by joeky888** [comment]
Hi @derekwyatt , would you provide your code as part of vim?

thanks ;)

see #1271

**Pull request closed without merging**